monocle/gitlab/mergerequest.py

```diff
@@ -0,0 +1,545 @@
+        kwargs = {
+            "mr_query": self.get_mr_query(include_commits=get_commits),
+            "url": self.url,
+            "group": self.group,
+            "repository": self.repository,
+            "updated_since": updated_since,
+            "after": "",
+            "total_mrs_count": 0,
+            "size": self.size,
+        }
+        one = 0
+        while True:
+            self.log.info(
+                "Running request %s"
+                % dict([(k, v) for k, v in kwargs.items() if k != "mr_query"])
+            )
+            try:
+                hnp = self._getPage(kwargs, mrs)
+                if kwargs["size"] == 1:
+                    self.log.debug(
+                        f"Getting this MR, with page size 1: {mrs[0]}"
+                    )
+                kwargs["size"] = min(
+                    MAX_BULK_SIZE, int(kwargs["size"] * AUGMENT) + 1
+                )
+                one = 0
+                if not get_commits:
+                    self.log.info("Will get full commits on next query.")
+                    kwargs["mr_query"] = self.get_mr_query(
+                        include_commits=get_commits
+                    )
+                    get_commits = True
+            except RequestTimeout:
+                kwargs["size"] = max(1, kwargs["size"] // REDUCE)
+                if kwargs["size"] == 1:
+                    one += 1
+                    if one == MAX_TRY - 1:
+                        self.log.info(
+                            "%d timeouts in a raw for one pr, retrying without commits."
+                            % (MAX_TRY - 1)
+                        )
+                        get_commits = False
+                        kwargs["mr_query"] = self.get_mr_query(
+                            include_commits=get_commits
+                        )
+                    elif one >= MAX_TRY:
+                        self.log.info(
+                            "%d timeouts in a raw for one pr, giving up."
+                            % MAX_TRY
+                        )
+                        raise
+                continue
+            self.log.info(f"{len(mrs)} MRs fetched")
+            if hnp is False:
+                break
+        return mrs
+
+    def get_one(self, url, group, repository, number):
+        def _dumper(raw, prefix=None):
+            pprint(raw)
+
+        qdata = """{
+          repository(owner: "%(group)s", name:"%(repository)s") {
+            mergeRequest(number: %(number)s) {
+              %(mr_query)s
+            }
+          }
+        }
+        """
+        kwargs = {
+            "mr_query": self.get_mr_query(),
+            "url": url,
+            "group": group,
+            "repository": repository,
+            "number": number,
+        }
+        raw = self.gql.query(qdata % kwargs)["data"]["repository"][
+            "pullRequest"
+        ]
+        return (raw, self.extract_objects([raw], _dumper))
+
+    def extract_objects(
```

Comment:
is this not used?
if this is used please comment and I will relook at it

Reply:
I don't know yet.  This function was taken from the dummy sample.